src/_http/http2Adapter.js

```diff
@@ -279,4 +290,17 @@ Http2Adapter.prototype.execute = function(options) {
   })
 }
 
+/**
+ * Destroys all the sessions and resets the session map.
+ *
+ * @returns {void}
+ */
+Http2Adapter.prototype.close = function() {
+  Object.values(this._sessionMap).forEach(function(sessionInterface) {
+    sessionInterface.destroy()
```

Comment:
Thanks for your help, greatly appreciate your input! You're totally right that it can be dangerous to destroy sessions expecting that the user will take care that there's no ongoing traffic and avoid any consequences caused by the possible interruption.
I've tested with the `.close` method and it works fine for regular requests (at least when they don't take much time to complete). But it hangs for streaming ones as expected, the script below won't be terminated:
```typescript
import { Client, query as Q } from 'faunadb'

const SECRET = process.env.FAUNA_SECRET_ADMIN_KEY!
const DOC_REF = Q.Ref(Q.Collection('items'), '295854750489903619')

const client = new Client({
  secret: SECRET,
  domain: 'db.fauna-preview.com',
  scheme: 'https',
})

async function main() {
  await new Promise<void>((resolve, reject) => {
    client.stream
      .document(DOC_REF)
      .on('snapshot', () => resolve())
      .on('error', reject)
      .start()
  })

  console.log("Listen to doc's changes")

  setTimeout(() => {
    console.log('Closing...')
    client.close()
  }, 2000)
}

main()
```

How do you think about `.close`ing regular requests and `.destroy`ing streaming ones? It may be some sort of compromise solution assuming that interruption of a streaming request isn't that crucial compared to a regular one which can perform a very important query.

We could also allow a parameter for the `Client#close` to leverage `.destroy` API instead of `.close`, e.g.:
```javascript
client.close({
  force: true, // Points to use session.destroy instead
})
```

Reply:
@retwere  I've pushed new changes to conform to what you've listed above, please check it out and let me know whether it makes sense.

Unfortunately, there is still a chance to hit the `ERR_HTTP2_GOAWAY_SESSION` error by calling the `.close` method and I have no idea what to do about it. The error may occur if the user calls `Client#close` too quickly (for me it's reproducible for 200ms delay) like on that simple code snippet we started our discussion from.
Here's a code snippet to reproduce on the latest PR's version:
```typescript
import { Client, query as Q } from 'faunadb'

const SECRET = process.env.FAUNA_SECRET_ADMIN_KEY!

const client = new Client({
  secret: SECRET,
  domain: 'db.fauna-preview.com',
  scheme: 'https',
})

function main() {
  client
    .query(Q.Now())
    .then(val => console.log('Response:', val))
    .catch(err => console.error('Failed:', err))

  setTimeout(() => {
    client.close()
  }, 200)
}

main()
```